fatal error: posix_types.h: No such file or directory in main

PR #79884 introduced build failure in main:

```
west build -p -b m2gl025_miv tests/posix/headers -T portability.posix.headers.without_posix_api
```

produces:

```
In file included from C:/work/zephyrproject/zephyr/lib/posix/options/device_io.c:14:
C:/work/zephyrproject/zephyr/include/zephyr/posix/sys/select.h:9:10: fatal error: posix_types.h: No such file or directory
    9 | #include "posix_types.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
```

--- a/include/zephyr/posix/dirent.h
+++ b/include/zephyr/posix/dirent.h
@@ -7,7 +7,8 @@
 #define ZEPHYR_INCLUDE_POSIX_DIRENT_H_
 
 #include <limits.h>
-#include "posix_types.h"
+
+#include <zephyr/posix/posix_types.h>
 
 #ifdef CONFIG_POSIX_FILE_SYSTEM
 #include <zephyr/fs/fs.h>

--- a/include/zephyr/posix/mqueue.h
+++ b/include/zephyr/posix/mqueue.h
@@ -12,7 +12,7 @@
 #include <zephyr/posix/fcntl.h>
 #include <zephyr/posix/signal.h>
 #include <zephyr/posix/sys/stat.h>
-#include "posix_types.h"
+#include <zephyr/posix/posix_types.h>
 
 #ifdef __cplusplus
 extern "C" {

--- a/include/zephyr/posix/posix_types.h
+++ b/include/zephyr/posix/posix_types.h
@@ -11,6 +11,18 @@
 #include <sys/types.h>
 #endif
 
+#if !defined(_CLOCK_T_DECLARED) && !defined(__clock_t_defined)
+typedef unsigned long clock_t;
+#define _CLOCK_T_DECLARED
+#define __clock_t_defined
+#endif
+
+#if !defined(_CLOCKID_T_DECLARED) && !defined(__clockid_t_defined)
+typedef unsigned long clockid_t;
+#define _CLOCKID_T_DECLARED
+#define __clockid_t_defined
+#endif
+
 #ifdef CONFIG_NEWLIB_LIBC
 #include <sys/_pthreadtypes.h>
 #endif

--- a/include/zephyr/posix/sched.h
+++ b/include/zephyr/posix/sched.h
@@ -7,8 +7,7 @@
 #define ZEPHYR_INCLUDE_POSIX_SCHED_H_
 
 #include <zephyr/kernel.h>
-
-#include "posix_types.h"
+#include <zephyr/posix/posix_types.h>
 
 #include <time.h>
 

--- a/include/zephyr/posix/semaphore.h
+++ b/include/zephyr/posix/semaphore.h
@@ -7,7 +7,7 @@
 #define ZEPHYR_INCLUDE_POSIX_SEMAPHORE_H_
 
 #include <zephyr/posix/time.h>
-#include "posix_types.h"
+#include <zephyr/posix/posix_types.h>
 
 #ifdef __cplusplus
 extern "C" {

--- a/include/zephyr/posix/signal.h
+++ b/include/zephyr/posix/signal.h
@@ -6,7 +6,8 @@
 #ifndef ZEPHYR_INCLUDE_POSIX_SIGNAL_H_
 #define ZEPHYR_INCLUDE_POSIX_SIGNAL_H_
 
-#include "posix_types.h"
+/* include posix_types.h before posix_features.h (here) to avoid build errors against newlib */
+#include <zephyr/posix/posix_types.h>
 #include "posix_features.h"
 
 #ifdef __cplusplus

--- a/include/zephyr/posix/sys/select.h
+++ b/include/zephyr/posix/sys/select.h
@@ -6,6 +6,7 @@
 #ifndef ZEPHYR_INCLUDE_POSIX_SYS_SELECT_H_
 #define ZEPHYR_INCLUDE_POSIX_SYS_SELECT_H_
 
+#include <zephyr/posix/posix_types.h>
 #include <zephyr/sys/fdtable.h>
 
 #ifdef __cplusplus

--- a/include/zephyr/posix/time.h
+++ b/include/zephyr/posix/time.h
@@ -58,7 +58,7 @@ struct itimerspec {
 
 #include <zephyr/kernel.h>
 #include <errno.h>
-#include "posix_types.h"
+#include <zephyr/posix/posix_types.h>
 #include <zephyr/posix/signal.h>
 
 #ifdef __cplusplus

--- a/include/zephyr/posix/unistd.h
+++ b/include/zephyr/posix/unistd.h
@@ -6,7 +6,7 @@
 #ifndef ZEPHYR_INCLUDE_POSIX_UNISTD_H_
 #define ZEPHYR_INCLUDE_POSIX_UNISTD_H_
 
-#include "posix_types.h"
+#include <zephyr/posix/posix_types.h>
 
 #ifdef CONFIG_POSIX_API
 #include <zephyr/fs/fs.h>

--- a/tests/posix/net/CMakeLists.txt
+++ b/tests/posix/net/CMakeLists.txt
@@ -7,3 +7,5 @@ project(posix_net)
 FILE(GLOB app_sources src/*.c)
 
 target_sources(app PRIVATE ${app_sources})
+
+target_compile_options(app PRIVATE -U_POSIX_C_SOURCE -D_POSIX_C_SOURCE=200809L)

